doc/ug/dv_methodology/index.md

```diff
@@ -465,7 +465,39 @@ It is recommended to follow these guidelines when collecting coverage:
+VCS UNR (Unreachability) is a formal solution that determines the unreachable coverage objects automatically from simulation.
+The same coverage hierarchy file and the exclusion files used for the simulation can be supplied to VCS UNR.
+
+Follow these steps to run and submit the exclusion file.
+1. Generate the VCS coverage database for the block by running full regression with `--cov` switch.
+2. Launch the VCS UNR flow:
+```
+util/dvsim/dvsim.py path/to/<dut>_sim_cfg.hjson --cov-unr
+```
+3. If no exclusion file is generated, there is no unreachable code in RTL.
+   If there is an exclusion file generated, the output should be reviewed by both designer and verification engineer.
+   When the unreachable code is sensible and we decide to exclude it in coverage report, create a PR to add to ['common_cov_excl.el'](https://github.com/lowRISC/opentitan/blob/master/hw/dv/tools/vcs/common_cov_excl.el) or block specific exclusion file, such as ['uart_cov_excl.el'](https://github.com/lowRISC/opentitan/blob/master/hw/ip/uart/dv/cov/uart_cov_excl.el)
+
+Here are some guidelines for using UNR and checking in generating exclusion.
+1. It's encouraged that designers run UNR to check the design in the early design stage (D1/D2), but adding exclusion for unreachable coverage should be done between D2 and V3 stage when the design is frozen (no feature update is allowed, except bug fix).
+Getting to 90% coverage via functional test is easy.
+Over 90% is the hard part as there may be a big chunk of unreachable codes.
+It is hard to go through coverage report to manually add exclusions, but VCS UNR flow provides a path to weed out all of unreachables ones.
+However, it is not the right thing to add a coverage exclusion file to reach the 80% needed for V2 since the design isn't stable at that period.
+2. If any RTL changes happen to the design after the coverage exclusion file has been created, it needs to be redone and re-reviewed.
+3. All coverage exclusion files and coverage configuration file (if it's not using default [cover.cfg](https://github.com/lowRISC/opentitan/blob/master/hw/dv/tools/vcs/cover.cfg)) should be checked during sign-off.
+4. Keep the VCS generated `CHECKSUM` along with exclusions, which is served as a crosscheck to ensure that the exclusion isn't applied when there is some change on the corresponding codes and the exclusion is outdated.
+We should not use `--exclude-bypass-checks` to disable the check, otherwise, it's needed to have additional review to make sure exclusions match to the design.
+5. For IP verified in IP-level test bench, UNR should be run in IP-level to generate exclusion.
+For IP verified in top-level, UNR should be run in top-level.
+There is no reuse of exclusions from IP to top, since they are independently closed for coverage.
+
+Note: VCS UNR doesn't support assertion or functional coverage.
```

Comment:
@weicaiyang I think this needs rewording.
it is not so much a lack of support. it simply does not make sense to talk about unreachable code when talking functional coverage.

```suggestion
Note: VCS UNR is only relevant when closing code coverage.
```

Reply:
I feel unreachable analysis is doable for assertion. FPV tool already can support it, but it's supported in VCS.
UNR for covergroup may be possible too in the future. I kind of prefer to explicitly mention what UNR can't support. WDYT?

Reply:
I think we are getting into semantics here.
unreachable functional coverage is to me a coverage hole

Reply:
Agree. usually it's just due to coding issue in TB, like forget to call the sample or assign to a wrong variable. I feel tool could help to sort that out.